www/js/goals.js

```diff
@@ -61,37 +81,334 @@ angular.module('emission.main.goals',['emission.services', 'ngSanitize'])
 		});
 		CommHelper.habiticaRegister(regConfig, function(response) {
 			console.log("Success!")
-			$scope.screen = response.success;
-			console.log(response);
-			$window.location.reload();
+			userInfo();
+			//console.log(response);
+			//$window.location.reload();
 		}, function(error) {
 			$ionicLoading.hide();
 			$ionicPopup.alert({title: "<h4 class='center-align'>Username is Required</h4>",
 								okText: "Try Again",
 								okType: "button-assertive"});
-			console.log(JSON.stringify(error));
+			console.log("Not signed up");
 		});
 	};
  
+
     $ionicLoading.show({
 			template: '<ion-spinner icon="bubbles" class="costume"></ion-spinner>'
 		});
 
-	var callOpts = {'method': 'GET', 'method_url': "/api/v3/user",
+
+	$scope.refreshUserInfo = function(){
+		console.log("Refreshing user information");
```

Comment:
This is really the only substantive comment for the PR. 
It looks like `refreshUserInfo` refreshes (duh) the user info.
But the bugs that we saw yesterday were around refreshing party state, party member list, challenge information, etc.

How do I refresh all of those?

My suggestion is to make this `refreshInfo` and have it call all the functions - `userInfo`, `userTask`, `getMembers`, etc. and then you can just call `refreshInfo()` when the page loads instead of calling `userInfo()`, `userTask()`, `getMembers()` etc